### PR TITLE
Make length() faster

### PR DESCRIPTION
**Pull request by kylekatarnls**

### CODE REVIEW

File: tests/BoundaryTest.php [hunk 1]
```diff
@@ -57,16 +58,69 @@ public function length_with_boundaries($expectedAmount, Period $period)
...
+
+            $iteratorTime -= microtime(true);
+            $iteratorCount = iterator_count($period);
+            $iteratorTime += microtime(true);
+
+            $calcTime -= microtime(true);
+            $length = $period->length();
+            $calcTime += microtime(true);
+
+            $this->assertEquals($iteratorCount, $length);
```
Comment (kylekatarnls) [inline tests/BoundaryTest.php]: Here we check `->length()` always returns the same as `iterator_count` would have returned.

File: tests/BoundaryTest.php [hunk 2]
```diff
@@ -57,16 +58,69 @@ public function length_with_boundaries($expectedAmount, Period $period)
...
+        }
+
+        $this->assertLessThan($iteratorTime, $calcTime);
```
Comment (kylekatarnls) [inline tests/BoundaryTest.php]: Here we check `->length()` is faster then `iterator_count()`

**Comment by brendt** [comment]
Thanks!

**Comment by brendt** [comment]
Tagged as 1.4.3: https://github.com/spatie/period/releases/tag/1.4.3

**Comment by kylekatarnls** [comment]
I PR it to 2.0 so it's not in 1.4.3: https://github.com/spatie/period/compare/1.4.2...1.4.3

**Comment by kylekatarnls** [comment]
For now in 1.4.3, it's optimized as length() is always returned as days count:
https://github.com/spatie/period/blob/1.4.3/src/Period.php#L138

But of course it does not match `iterator_count` as it ignore the precision mask.

I would say, as a breaking change, this should not be in 1.x.

**Comment by brendt** [comment]
Oh, you're right.. My bad. No problem though, we'll merge it in when 2.0 arrives

**Pull request merged**